### PR TITLE
D8 nid 290 social sharing tools

### PR DESCRIPTION
**Pull request by johangant**
Simple module to expose a block that can be put into the menu/footer as needed, rather than having to use Drupal's menu system to do precisely the same thing with less control

### CODE REVIEW

File: nidirect_sst/nidirect_sst.module [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Contains nidirect_sst.module.
+ */
```
Comment (DuttonMa) [inline nidirect_sst/nidirect_sst.module]: As this file is empty, maybe you should just get rid of it ?

File: nidirect_sst/src/Plugin/Block/SharingLinks.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = [];
+
+    // Links for: Twitter, Facebook, Youtube and RSS.
+    $build['twitter'] = Link::fromTextAndUrl($this->t('Twitter'),
+      Url::fromUri('https://twitter.com/nidirect', [
+        'attributes' => ['title' => $this->t('Twitter')]
+      ]))->toRenderable();
```
Comment (DuttonMa) [inline nidirect_sst/src/Plugin/Block/SharingLinks.php]: I like this construction - I must start building my links this way !

**Comment by johangant** [comment]
Parking this PR for the moment. @omahm suggested some Twig template links as a first step, with the possibility of config from the theme layer if it needs to be re-used or re-configured with minimal fuss.

File: nidirect_sst/nidirect_sst.module (hunk 1 above)
Comment (johangant) [reply nidirect_sst/nidirect_sst.module]: Agreed, didn't spot this. Thanks

**Comment by johangant** [comment]
So it's entirely possible to add the render array elements from the block implementation directly into the theme region render array, although Bartik's default implementation is a tad simplistic (prints a pre-rendered HTML variable) but the nidirect theme could easily render `content.sst_links` or use preprocess_page to squirt in the relevant render elements.

I'm double-checking consensus on how best to present this stuff to avoid having to unpick/redo later.

**Comment by johangant** [comment]
Marking as prospective future enhancement and closing.

**Pull request closed without merging**